src/library-authoring/components/PublishConfirmationModal.tsx

```diff
@@ -0,0 +1,83 @@
+import { useEffect } from 'react';
+import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
+import { Alert, Button, useToggle } from '@openedx/paragon';
+
+import BaseModal from '../../editors/sharedComponents/BaseModal';
+import messages from './messages';
+import infoMessages from '../component-info/messages';
+import { ComponentUsage } from '../component-info/ComponentUsage';
+
+interface PublishConfirmationModalProps {
+  isOpen: boolean,
+  onClose: () => void,
+  onConfirm: () => void,
+  displayName: string,
+  usageKey: string,
+  showDownstreams: boolean,
+}
+
+const PublishConfirmationModal = ({
+  isOpen,
+  onClose,
+  onConfirm,
+  usageKey,
+  displayName,
+  showDownstreams,
+}: PublishConfirmationModalProps) => {
+  const intl = useIntl();
+
+  const [
+    isDownstreamsEmpty,
+    setDownstreamsEmpty,
+    setDownstreamsNotEmpty,
+  ] = useToggle(false);
+
+  useEffect(() => {
+    // Set to default 'false' when changing usage key
+    setDownstreamsNotEmpty();
+  }, [usageKey]);
```

Comment:
This pattern of using a toggle for `isDownstreamsEmpty` and a `callbackEmpty()` callback on the child component is not really the way React is supposed to work.

Can you just put something like this, to load the same data directly and use it? React Query will de-duplicate the requests so there's no additional overhead, and it makes the code much simpler.

```suggestion
const {
    data: dataDownstreamLinks,
    isLoading: isLoadingDownstreamLinks,
  } = useComponentDownstreamLinks(usageKey);
const hasNoDownstreamUsages = !isLoadingDownstreamLinks && dataDownstreamLinks.length == 0;
```

Reply:
Thanks for the recommendation. Update https://github.com/openedx/frontend-app-authoring/pull/1677/commits/4fa60a66ca981eff91bf464ccaf31c2445e4ad6e